### ISSUE TITLE
1.8-alpha2 - istio-proxy cannot start when following multi-cluster install

### ISSUE THREAD

**Issue by marcfong** [istio/istio#28616]
**Bug description**
istio-proxy of the helloworld sample/validation app cannot start when following multi-cluster install.
https://preliminary.istio.io/latest/docs/setup/install/multicluster/multi-primary_multi-network/

[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**
helloworld pod can start up.

**Steps to reproduce the bug**
1. Testing done in GKE on GCP, two different projects, different VPC networks peered together.
2. Follow install steps from https://preliminary.istio.io/latest/docs/setup/install/multicluster/multi-primary_multi-network/
3. Follow verification steps from https://preliminary.istio.io/latest/docs/setup/install/multicluster/verify/

Observe the following error logs in istio-proxy
```
2020-11-05T14:47:07.490711Z	info	cache	Loaded root cert from certificate ROOTCA
2020-11-05T14:47:07.491026Z	info	sds	resource:ROOTCA pushed root cert to proxy
2020-11-05T14:47:07.508548Z	warning	envoy config	gRPC config for type.googleapis.com/envoy.config.listener.v3.Listener rejected: Proto constraint validation failed (ListenerValidationError.FilterChains[i]: ["embedded message failed validation"] | caused by FilterChainValidationError.Filters[i]: ["embedded message failed validation"] | caused by FilterValidationError.Name: ["value length must be at least " '\x01' " runes"]): name: "10.216.11.120_15443"
address {
  socket_address {
    address: "10.216.11.120"
    port_value: 15443
  }
}
filter_chains {
  filters {
  }
  filters {
    name: "envoy.filters.network.tcp_proxy"
    typed_config {
      [type.googleapis.com/envoy.extensions.filters.network.tcp_proxy.v3.TcpProxy] {
        stat_prefix: "outbound|15443||istio-ingressgateway.istio-system.svc.cluster.local"
        cluster: "outbound|15443||istio-ingressgateway.istio-system.svc.cluster.local"
      }
    }
  }
}
deprecated_v1 {
  bind_to_port {
  }
}
traffic_direction: OUTBOUND

2020-11-05T14:47:08.551257Z	warn	Envoy proxy is NOT ready: config not received from Pilot (is Pilot running?): cds updates: 1 successful, 0 rejected; lds updates: 0 successful, 1 rejected
```

Observe the following error logs in istiod
```
2020-11-05T14:31:54.324365Z     warn    ads     ADS:LDS: ACK ERROR sidecar~10.212.3.5~helloworld-v1-df6d74ccc-l4dwv.sample~sample.svc.cluster.local-6 Internal:Proto constraint validation failed (ListenerValidationError.FilterChains[i]: ["embedded message failed validation"] | caused by FilterChainValidationError.Filters[i]: ["embedded message failed validation"] | caused by FilterValidationError.Name: ["value length must be at least " '\x01' " runes"]): name: "10.216.6.65_15012"
```

**Version (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version` if you used Helm)**
client version: 1.8.0-alpha.2
control plane version: 1.8.0-alpha.2
data plane version: 1.8.0-alpha.2 (3 proxies)

**How was Istio installed?**

istioctl

**Environment where bug was observed (cloud vendor, OS, etc)**
GKE on GCP

**Comment by howardjohn** [comment]
Likely fixed by #28439, asked offline to try head of 1.8

**Comment by richardwxn** [comment]
It is fixed by #28439, the in-place upgrade test and canary upgrade test pass since then:  

1. https://prow.istio.io/job-history/gs/istio-prow/logs/istio-dual-control-plane-upgrade-1.7-1.8.
2. https://prow.istio.io/job-history/gs/istio-prow/logs/istio-upgrade-using-istioctl-1.7_latest-1.8_latest

I guessed it was caused by the bad CRDs/CRs hanging in the cluster, you probably did not clean up the CRD. can you get a dump of the envoyfilter CRD and one of the CRs so I can check the history.

**Comment by marcfong** [comment]
purging the CRDs and installing build `1.8-alpha.09df30801a3b1bc4a6c3e2dde6fb7a633003741f` solved the issue
